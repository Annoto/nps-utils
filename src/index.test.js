import path from 'path'
import {concurrent, series, runInNewWindow} from '.'

test('series', () => {
  expect(
    series('echo hey', null, 'echo hi', undefined, 'echo there', false),
  ).toMatchSnapshot()
})

test('series.nps', () => {
  expect(
    series.nps(
      'test',
      false,
      'lint.src',
      null,
      'lint.scripts --cache',
      undefined,
      ' ',
      'build --fast',
    ),
  ).toMatchSnapshot()
})

test('concurrent', () => {
  expect(
    concurrent({
      test: 'echo test',
      validate: {
        script: false,
      },
      lint: {script: 'echo lint', color: 'bgWhite.black.dim'},
      build: false,
      cover: undefined,
    }),
  ).toMatchSnapshot()
})

test('concurrent.nps', () => {
  expect(
    concurrent.nps(
      null,
      'test',
      undefined,
      'lint',
      {script: 'build.app --silent'},
      false,
      {script: 'validate', color: 'bgGreen.dim'},
    ),
  ).toMatchSnapshot()
})

test('runInNewWindow', () => {
  expect(relativeizePath(runInNewWindow('echo hi'))).toMatchSnapshot()
})

test('runInNewWindow.nps as windows', () => {
  const originalPlatform = process.platform
  process.platform = 'win32'
  jest.resetModules()
  const freshUtils = require('.')
  expect(
    relativeizePath(freshUtils.runInNewWindow.nps('initiate database')),
  ).toMatchSnapshot()
  process.platform = originalPlatform
})

function relativeizePath(stringWithAbsolutePaths) {
  return stringWithAbsolutePaths.replace(
    new RegExp(path.resolve(__dirname, '../'), 'g'),
    '<projectRootDir>',
  )
}

/*
  eslint
    global-require:0,
    import/no-dynamic-require:0
*/
