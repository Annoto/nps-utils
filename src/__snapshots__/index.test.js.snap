// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`concurrent 1`] = `"node node_modules/concurrently/src/main.js --kill-others-on-fail --prefix-colors \\"bgBlue.bold,bgWhite.black.dim\\" --prefix \\"[{name}]\\" --names \\"test,lint\\" 'echo test' 'echo lint'"`;

exports[`concurrent.nps 1`] = `"node node_modules/concurrently/src/main.js --kill-others-on-fail --prefix-colors \\"bgMagenta.bold,bgBlack.bold,bgCyan.bold,bgGreen.dim\\" --prefix \\"[{name}]\\" --names \\"test,lint,build.app,validate\\" 'nps test' 'nps lint' 'nps \\"build.app --silent\\"' 'nps validate'"`;

exports[`runInNewWindow 1`] = `"osascript -e 'tell application \\"Terminal\\"' -e 'tell application \\"System Events\\" to keystroke \\"t\\" using {command down}' -e 'do script \\"cd <projectRootDir> && echo hi\\" in front window' -e 'end tell'"`;

exports[`runInNewWindow.nps as windows 1`] = `"start cmd /k \\"cd <projectRootDir> && node node_modules/.bin/nps \\\\\\"initiate database\\\\\\"\\""`;

exports[`series 1`] = `"echo hey && echo hi && echo there"`;

exports[`series.nps 1`] = `"nps test && nps lint.src && nps \\"lint.scripts --cache\\" && nps \\"build --fast\\""`;
